flyteadmin/pkg/errors/errors.go

```diff
@@ -109,8 +111,47 @@ func NewIncompatibleClusterError(ctx context.Context, errorMsg, curCluster strin
 	return statusErr
 }
 
-func NewWorkflowExistsDifferentStructureError(ctx context.Context, request *admin.WorkflowCreateRequest) FlyteAdminError {
-	errorMsg := "workflow with different structure already exists"
+func compareJsons(jsonArray1 jsondiff.Patch, jsonArray2 jsondiff.Patch) []string {
+	results := []string{}
+	map1 := make(map[string]jsondiff.Operation)
+	for _, obj := range jsonArray1 {
+		map1[obj.Path] = obj
+	}
+
+	for _, obj := range jsonArray2 {
+		if val, ok := map1[obj.Path]; ok {
+			result := fmt.Sprintf("\t\t- %v: %v -> %v", obj.Path, obj.Value, val.Value)
+			results = append(results, result)
+		}
+	}
+	return results
+}
+
+func NewTaskExistsDifferentStructureError(ctx context.Context, request *admin.TaskCreateRequest, oldSpec *core.CompiledTask, newSpec *core.CompiledTask) FlyteAdminError {
+	errorMsg := "task with different structure already exists:\n"
+	diff, _ := jsondiff.Compare(oldSpec, newSpec)
```

Comment:
since in golang map ordering is random/never guaranteed, I'm worried that converting the specs to bytes here might lead to non deterministic byte arrays

when we compute the digest, we use [pbhash](https://github.com/flyteorg/flyte/blob/2adaf612348c9b3d83f4640b0eedba6e2ab33f4c/flyteadmin/pkg/manager/impl/util/digests.go#L30) to marshal the protos to json 

should we do the same here before calling Compare()?

alternatively, do you mind double checking that jsondiff.Compare() returns no diff for multiple calls comparing 2 identical pb objects that contain maps in the message?

Reply:
can!
I've carefully double checked that `jsondiff.Compare()` is a identical operation. Also added the test for this function.

Reply:
thanks @austin362667 for the super fast turnaround!

in this case, i was more concerned about identical maps not showing a false diff - would you mind double checking that? otherwise PR looks great!!

Reply:
done